src/nvim/globals.h

```diff
@@ -746,6 +746,9 @@ EXTERN bool enc_utf8 INIT(= false);             /* UTF-8 encoded Unicode */
 EXTERN int enc_latin1like INIT(= TRUE);         /* 'encoding' is latin1 comp. */
 EXTERN int has_mbyte INIT(= 0);                 /* any multi-byte encoding */
 
+// 'encoding' can only be changed in the startup scripts
+EXTERN bool enc_mutable INIT(= true);
```

Comment:
> Why not just rename enc_mutable to startupscripts_loaded or something

Ok. The convention in globals.h seems to be `did_foo`, so how about `did_source_startup_scripts` and set it at the end of `source_startup_scripts()`.

I was wrong about `current_SID` by the way, it is 0 unless a script is running.